docs/criufeature.md

```diff
@@ -0,0 +1,93 @@
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# CRIU feature
+
+**(Linux&reg; x86, Linux on POWER LE&reg;, and Linux on IBM Z&reg; only)**
+
+:fontawesome-solid-pencil:{: .note aria-hidden="true"} **Note:** CRIU support is a technical preview feature that is intended for container environments. All APIs and options are subject to change. CRIU support requires the version of CRIU that is a package in Semeru UBI container images.
+
+In Eclipse OpenJ9&trade;, Checkpoint Restore in User Space (CRIU) feature is available as an API that can be used to stop the VM at a particular checkpoint, save the state, and then enable the VM to resume from the point where it was stopped. The time between the VM start and checkpoint is referred to as the checkpoint phase. The checkpointed state is written to one or more image files with the memory content available during the checkpoint phase, such as memory pages, methods, file systems, processes, network connections, and security components. You can restore the checkpoint image from these files in as many environments as possible with improved start-up time and hence with better VM performance.
+
+For more information about the CRIU feature and OpenJ9 CRIU support, see the following blog posts:
+
+- [Fast JVM startup with OpenJ9 CRIU Support](https://blog.openj9.org/2022/09/26/fast-jvm-startup-with-openj9-criu-support/)
+- [Getting started with OpenJ9 CRIU Support](https://blog.openj9.org/2022/09/26/getting-started-with-openj9-criu-support/)
+- [OpenJ9 CRIU Support: A look under the hood](https://blog.openj9.org/2022/10/14/openj9-criu-support-a-look-under-the-hood/)
+- [OpenJ9 CRIU Support: A look under the hood (part II)](https://blog.openj9.org/2022/10/14/openj9-criu-support-a-look-under-the-hood-part-ii/)
+
+## Using the CRIU feature
+
+CRIU feature is not enabled by default, you must explicitly invoke it. You can invoke the feature with the `-XX:+EnableCRIUSupport` option. When thus enabled, the VM starts with support for checkpoint and restore. The VM will launch the JIT, GC, Security, and other internal components that make checkpoint and restore feasible.
+
+For more information about the `-XX:+EnableCRIUSupport` option, see the [`-XX:[+|-]EnableCRIUSupport`](xxenablecriusupport.md) topic.
+
+## Configuring the CRIU support feature
+
+You can use command line options to configure the CRIU support to use CRIU capabilities that are provided by the operating system as well as VM support.
+
+### New options for checkpoint and restore process
```

Comment:
The entries in the right-hand toc are too long and are running off the page. I would:
- Add some sentences before this heading to explain more, eg that there are some new options, that some can be used during the checkpoint phase only and some after the restore operation, that not all existing options can be used after a restore operation, and that some of those that can behave differently. (There are a *lot* of command-line options that aren't mentioned in this topic - do none of them work after restore? What happens if a user tries to use one, like -Xsyslog for example?)
- Change this heading to "New options" then use lead-in sentences to separate the checkpoint phase options from the post-restore options
- Remove the enabling option since it was introduced in the previous section and isn't about configuring
- Change the next heading to "Existing options" and add a lead-in sentence for explanation, eg "Of the existing command-line options, only the following are supported when you run a restored VM."
- Use colons instead of dashes
- I think we should list the existing options as they are in the issue, ie adding extra explanation only if the behaviour is different from normal. Otherwise, the reader can get the impression that the description for other options is different behaviour too.

I had suggested trying a table but a list is at least consistent with JITServer doc.

Reply:
@tajila - In the issue, a list of existing options that work on restore is given. So, does that mean that all the other existing command-line options will not work on restore? 

What happens if a user tries to use one, like -Xsyslog for example?

Reply:
> So, does that mean that all the other existing command-line options will not work on restore?

yes

> What happens if a user tries to use one, like -Xsyslog for example?

By default, specifying an unsupported option will cause the JVM to throw a JVMRestoreException. If `-XX:+IgnoreUnrecognizedRestoreOptions` no exception is thrown and the JVM just ignores unsupported options.